src/frontend/test_runner/tests/testdata/join.yaml

```diff
@@ -507,6 +536,27 @@
                 StreamTableScan { table: a, columns: [a.x, a._row_id], pk: [a._row_id], distribution: HashShard(a._row_id) }
               StreamExchange { dist: HashShard(b.x) }
                 StreamTableScan { table: b, columns: [b.x, b._row_id], pk: [b._row_id], distribution: HashShard(b._row_id) }
+- sql: |
+    /* Test that we can join subqueries without aliases */
+    select * from (select 1 as x) join (select 1 as x) using(x);
```

Comment:
Subqueries without aliases seems not a correct syntax？

Reply:
At least, both MySQL and PostgreSQL don't consider this sql as a valid sql.
They both think every derived table must have its own alias.

Reply:
Yes, it's not supported in those. 

Maybe I'll post in the slack channel to ask what people think we should follow. But for instance values also has no alias, and at least for now I don't see a practical reason why we should not support subquery in from statement without alias.

Reply:
But anyway, it's not a problem not to support it either. I can remove this test and in another PR we can enforce that FROM subquery always has an alias in parser